docs/tutorials/weaviate.md

```diff
@@ -0,0 +1,227 @@
+# Weaviate with local autoscaling embedding and generative models
+
+Weaviate is a vector search engine that can integrate seamlessly with KubeAI's embedding and generative models. This tutorial demonstrates how to deploy both KubeAI and Weaviate in a Kubernetes cluster, using KubeAI as the OpenAI endpoint for Weaviate.
+
+This tutorial uses CPU only models, so it should work even on your laptop.
+
+As you go go through this tutorial, you will learn how to:
+- Deploy KubeAI with embedding and generative models
+- Install Weaviate and connect it to KubeAI
+- Import data into Weaviate
+- Perform semantic search using the embedding model
+- Perform generative search using the generative model
+
+## Prerequisites
+A Kubernetes cluster. You can use [kind](https://kind.sigs.k8s.io/) or [minikube](https://minikube.sigs.k8s.io/docs/start/).
+
+```bash
+kind create cluster
+```
+
+## KubeAI Configuration
+Let's start by deploying KubeAI with the models we want to use.
+Nomic embedding model is used instead of text-embedding-ada-002.
+Gemma 2 2B is used instead of gpt-3.5-turbo.
+You could choose to use bigger models depending on your available hardware.
+
+Create a file named `kubeai-values.yaml` with the following content:
+```yaml
+models:
+  catalog:
+    text-embedding-ada-002:
+      enabled: true
+      minReplicas: 1
+      features: ["TextEmbedding"]
+      owner: nomic
+      url: "ollama://nomic-embed-text"
+      engine: OLlama
+      resourceProfile: cpu:1
+    gpt-3.5-turbo:
+      enabled: true
+      minReplicas: 1
+      features: ["TextGeneration"]
+      owner: google
+      url: "ollama://gemma2:2b"
+      engine: OLlama
+      resourceProfile: cpu:2
+```
+
+Note: It's important that you name the models as `text-embedding-ada-002` and `gpt-3.5-turbo` as Weaviate expects these names.
+
+Run the following command to deploy KubeAI:
+```bash
+helm upgrade --install kubeai kubeai/kubeai \
+    -f ./kubeai-values.yaml --reuse-values
```

Comment:
No need for `--reuse-values` here.

Reply:
Hmm I kept it so people don't accidentally delete their existing models.